tests/nosetests/pyanaconda_tests/module_payload_dnf_test.py

```diff
@@ -146,6 +147,166 @@ def url_metalink_kickstart_test(self):
         self.shared_ks_tests.check_kickstart(ks_in, ks_out)
         self._check_properties(SOURCE_TYPE_URL)
 
+    def packages_section_empty_kickstart_test(self):
+        """Test the empty packages section."""
+        ks_in = """
+        %packages
+        %end
+        """
+        ks_out = """
+        %packages
+
+        %end
+        """
+        self.shared_ks_tests.check_kickstart(
+            ks_in, ks_out="", ks_tmp=ks_out
+        )
+
+    def packages_section_kickstart_test(self):
+        """Test the packages section."""
+        ks_in = """
+        %packages
+        package
+        @group
+        @module:10
+        @module2:1.5/server
+        @^environment
+        %end
+        """
+        ks_out = """
+        %packages
+        @^environment
+        @group
+        @module2:1.5/server
+        @module:10
+        package
+
+        %end
+        """
+        self.shared_ks_tests.check_kickstart(
+            ks_in, ks_out="", ks_tmp=ks_out
+        )
+
+    def packages_section_complex_kickstart_test(self):
+        """Test the packages section with duplicates."""
+        ks_in = """
+        %packages
+        @^environment1
+        package1
+        @group1 --nodefaults
+        package2
+
+        # Only this environment will stay (last specified wins)
+        @^environment2
+        @group2
+        @group3 --optional
+
+        # duplicates
+        package2
+        @group2
+
+        # modules
+        @module:4
+        @module:3.5/server
+
+        %end
+        """
+        # The last specified environment wins, you can't specify two environments
+        # Same package or group specified twice will be deduplicated
+        ks_out = """
+        %packages
+        @^environment2
+        @group1 --nodefaults
+        @group2
+        @group3 --optional
+        @module:3.5/server
+        @module:4
+        package1
+        package2
+
+        %end
+        """
+        self.shared_ks_tests.check_kickstart(
+            ks_in, ks_out="", ks_tmp=ks_out
+        )
+
+    def packages_section_with_attribute_kickstart_test(self):
```

Comment:
What does this do that the next test does not?

Reply:
I guess the case of just `--nocore` without anything else ?

Reply:
The tests are reorganized a little.